### PR TITLE
Single subtab in memory

### PR DESCRIPTION
**Pull request by evanweaver**
@kbenne thanks for looking this over.

### CODE REVIEW

File: openstudiocore/src/openstudio_lib/ConstructionsTabController.cpp [hunk 1]
```diff
@@ -32,29 +32,80 @@
 namespace openstudio {
 
 ConstructionsTabController::ConstructionsTabController(bool isIP, const model::Model& model)
-  : MainTabController(new ConstructionsTabView(model, "Constructions"))
+  : MainTabController(new ConstructionsTabView(model, "Constructions")),
+  m_model(model),
+  m_isIP(isIP)
 {
```
Comment (kbenne) [inline openstudiocore/src/openstudio_lib/ConstructionsTabController.cpp]: Are these controllers actually removed? that would be awesome!

File: openstudiocore/src/openstudio_lib/ConstructionsTabController.cpp [hunk 2]
```diff
@@ -32,29 +32,80 @@
...
+{
+  if (m_currentIndex == index) {
+    return;
+  }
+  else {
+    m_currentIndex = index;
+  }
 
-  connect(this, &ConstructionsTabController::toggleUnitsClicked, static_cast<ModelSubTabView*>(m_materialsController->subTabView()), &ModelSubTabView::toggleUnitsClicked);
+  if (qobject_cast<DefaultConstructionSetsController *>(m_currentController)) {
```
Comment (kbenne) [inline openstudiocore/src/openstudio_lib/ConstructionsTabController.cpp]: yeah ok this is all weird, but i can't argue with improvement. I would still like to understand better.

File: openstudiocore/src/openstudio_lib/ConstructionsTabController.cpp [hunk 3]
```diff
@@ -32,29 +32,80 @@
...
+  else if (m_currentController) {
+    // Oops! Should never get here
+    OS_ASSERT(false);
+  }
 
-  connect(m_defaultConstructionSetsController.get(), &DefaultConstructionSetsController::downloadComponentsClicked, this, &ConstructionsTabController::downloadComponentsClicked);
+  switch (index){
+  case 0:
+  {
+    auto defaultConstructionSetsController = new DefaultConstructionSetsController(m_model);
```
Comment (kbenne) [inline openstudiocore/src/openstudio_lib/ConstructionsTabController.cpp]: I see they aren't gone. just created on demand. aka single tab in memory. doh!

Comment (evanweaver) [inline openstudiocore/src/openstudio_lib/ConstructionsTabController.cpp]: None the less, it was exciting for a moment.

**Pull request merged**